### PR TITLE
[chart] Use bitnami/common function + provide compatibilty with different K8s versions

### PR DESCRIPTION
**Pull request by juan131**
Signed-off-by: juan131 <juanariza@vmware.com>

### Description of the change

This PR adapts the Kubeapps chart to use the `bitnami/common` library so several common macros are reused. This is important to standardize this chart with the rest of the Bitnami Catalog.

On the other hand, it ensures the `apiVersion` of several objects (e.g. Ingress, RBAC, CRDs, etc.) is compatible with different K8s versions.

### Benefits

- Standardization
- Compatibility with multiple K8s versions.

### Possible drawbacks

None

### Applicable issues

N/A

### Additional information

None

### CODE REVIEW

File: chart/kubeapps/crds/apprepository-crd.yaml
Comment (absoludity) [inline chart/kubeapps/crds/apprepository-crd.yaml:1]: Hrm. I thought we (bitnami) were trying to move away from helm functionality which moves away from `constant-input  + chart = constant-output`? That is, using `.Capabilities` (which this does under the hood), or `lookup` functions (not used here, just as example, though I see it is used in the `_secrets.tpl`) means that the output is a function of not only the input and chart, but also the current cluster context.

Not blocking on this as I assume the decision's been made already. We do already have one use of `.Capabilities` but were trying not to add more.

**Review by absoludity — COMMENTED** [review]
Thanks @juan131 for all the work updating that - and removing the no-longer needed kubeapps helpers!
I'm a little worried that we're (generally) adding more helm specific functionality that moves away from the resulting templates a function of the input and chart only (ie. relying on current k8s environment), but that's not really relevant for this PR.

I'll hold off approving until both @andresmgot 's #2436 is landed (has the same chart version) and the e2e test is passing. It's currently failing with an error that looks like it could be related. I've just re-run the failed test to be sure, but haven't looked more closely.

File: chart/kubeapps/templates/NOTES.txt
Comment (antgamdia) [inline chart/kubeapps/templates/NOTES.txt:9]: So, from now on, we shouldn't be worried about the length of the name anymore since this function is automatically truncating it, isn't it?

File: chart/kubeapps/templates/ingress.yaml
Comment (antgamdia) [inline chart/kubeapps/templates/ingress.yaml:32]: Not a chart expert at all. However, I think there are more changes here if we use `networking.k8s.io/v1` instead of `extensions/v1beta1`: https://github.com/kubeapps/kubeapps/issues/2169#issuecomment-733556537

File: chart/kubeapps/crds/apprepository-crd.yaml
Comment (antgamdia) [inline chart/kubeapps/crds/apprepository-crd.yaml:13]: I guess here we are solving this: https://github.com/kubeapps/kubeapps/issues/2169#issuecomment-728910010
No?

File: chart/kubeapps/Chart.yaml
Comment (andresmgot) [reply chart/kubeapps/Chart.yaml:28]: Actually I am going to release 5.2.2 soon so this should be 5.2.3

**Review by antgamdia — COMMENTED** [review]
I agree with Michael's comments. We are relying too much on the helm and server context. It may harden the migration from Help to other packaging formats. 
However, from a Bitnami chart perspective, it's a great change since API versions will automagically adapt to the proper ones.

Thanks for the effort and sending this PR :)

File: chart/kubeapps/crds/apprepository-crd.yaml
Comment (andresmgot) [reply chart/kubeapps/crds/apprepository-crd.yaml:1]: Not sure about this, but I guess the goal is to define `.Values.global.kubeVersion` so `.Capabilities` is not called. Can you confirm @juan131 ?

File: chart/kubeapps/templates/apprepository-rbac.yaml
Comment (andresmgot) [inline chart/kubeapps/templates/apprepository-rbac.yaml:95]: have you removed the `kind` by accident?

Comment (andresmgot) [reply chart/kubeapps/templates/apprepository-rbac.yaml:95]: this seems why the CI is failing:

```
Error: UPGRADE FAILED: error validating "": error validating data: kind not set
```

Comment (juan131) [reply chart/kubeapps/templates/apprepository-rbac.yaml:95]: Correct. I'll fix it

File: chart/kubeapps/Chart.yaml
Comment (juan131) [reply chart/kubeapps/Chart.yaml:28]: No worries. I'll rebase and update it.

File: chart/kubeapps/crds/apprepository-crd.yaml
Comment (juan131) [reply chart/kubeapps/crds/apprepository-crd.yaml:1]: Correct. Let me added the missing values and a couple of refactorizations related to `bitnami/common` more

File: chart/kubeapps/templates/NOTES.txt
Comment (juan131) [reply chart/kubeapps/templates/NOTES.txt:9]: Yup, it works pretty similar to what you had before.

File: chart/kubeapps/templates/ingress.yaml
Comment (juan131) [reply chart/kubeapps/templates/ingress.yaml:32]: Yes, let me the rest of refactorization for ingress.

File: chart/kubeapps/crds/apprepository-crd.yaml
Comment (juan131) [reply chart/kubeapps/crds/apprepository-crd.yaml:13]: I guess, but it'd be nice if you give that a try

Comment (juan131) [reply chart/kubeapps/crds/apprepository-crd.yaml]: Reverting these changes, since only pure YAML (no templates) is accepted for CRDs

File: chart/kubeapps/templates/tls-secrets.yaml
Comment (andresmgot) [inline chart/kubeapps/templates/tls-secrets.yaml:31]: I think you have an extra `end` here, I only see one `if` opening here

**Review by andresmgot — COMMENTED** [review]
Can you take a look to the comment?

This PR is too big to be reviewed at once. Can you avoid the version bump? At least we would need to do some manual testing before releasing the version.

File: chart/kubeapps/templates/tls-secrets.yaml
Comment (juan131) [reply chart/kubeapps/templates/tls-secrets.yaml:31]: You're missing the `range` that's also finished with an `end`. This `end` is necessary.

**Comment by juan131** [comment]
> This PR is too big to be reviewed at once. Can you avoid the version bump? At least we would need to do some manual testing before releasing the version.

Take your time to test it, and ping me every time you consider it necessary. I do consider the version bump necessary since new parameters are included in the chart (to name a few: `ingress.path`, `ingress.pathType`,  `kubeVersion`, etc.`)

File: chart/kubeapps/templates/tls-secrets.yaml
Comment (andresmgot) [reply chart/kubeapps/templates/tls-secrets.yaml:31]: ah, got it, yes (I miss indentation at least)

**Comment by andresmgot** [comment]
> @andresmgot take your time to test it, and ping me every time you consider it necessary. I do consider the version bump necessary since new parameters are included in the chart (to name a few: `ingress.path`, `ingress.pathType`, `kubeVersion`, etc.)

Yes, I didn't mean that the version bump was wrong but I was thinking on testing master. In any case it's not necessary, so yes, I will give this a try tomorrow :+1: Thanks!

**Review by andresmgot — APPROVED** [review]
Sorry for the very late response!

I have tested this a little bit, setting different ingress values and setting the `kubeVersion`. Everything works as expected. Thanks!

**Pull request merged**
